pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/codearea/StyleContext.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.util.codearea;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.fxmisc.richtext.model.StyleSpans;
+import org.fxmisc.richtext.model.StyleSpansBuilder;
+
+/**
+ * Stores the current style layers and can overlay them into a {@link StyleSpans} to style the text.
+ */
+class StyleContext implements Iterable<StyleLayer> {
```

Comment:
not public?

Reply:
Yes. There's no use case for it outside the internals of the code area. The methods could be public though, I'll change that